nf_core/components/create.py

```diff
@@ -411,3 +430,131 @@ def _get_username(self):
                 f"[violet]GitHub Username:[/]{' (@author)' if author_default is None else ''}",
                 default=author_default,
             )
+
+    def _copy_old_files(self, component_old_path):
+        """Copy files from old module to new module"""
+        log.debug("Copying original main.nf file")
+        shutil.copyfile(component_old_path / "main.nf", self.file_paths[self.component_type + "/main.nf"])
```

Comment:
Comments log to not lose track:

[mashehu](https://github.com/mashehu) [2 days ago](https://github.com/nf-core/tools/pull/2542#discussion_r1406436826)
why not self.component_dir as the target for the main.nf?

@[mirpedrol](https://github.com/mirpedrol) mirpedrol [yesterday](https://github.com/nf-core/tools/pull/2542#discussion_r1407981536)
We are using the paths in file_paths which use self.directory.
Actually I find self.component_dir very misleading because it's the tool name (with /), not a path

@[mashehu](https://github.com/mashehu) mashehu [13 minutes ago](https://github.com/nf-core/tools/pull/2542#discussion_r1409220269)
oh, yes that's bad. maybe if we initialize the type as string for this variable it becomes clearer when we make this mistake? other option: rename it to component_dir_name

Reply:
Addressed in https://github.com/nf-core/tools/pull/2551